### ISSUE TITLE
az bicep build broken in CLI 2.22.0

### ISSUE THREAD

**Issue by IJsbrandVV** [Azure/azure-cli#17665]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
Running `az bicep build` gives a python error:

```
The command failed with an unexpected error. Here is the traceback:
type object 'datetime.datetime' has no attribute 'fromisoformat'
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 231, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 657, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 691, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 328, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/__init__.py", line 807, in default_command_handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/custom.py", line 3294, in build_bicep_file   
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/_bicep.py", line 63, in run_bicep_command    
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/_bicep.py", line 152, in _load_bicep_version_check_result_from_cache
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'
```

**To Reproduce**

- Upgrade to Azure CLI 2.22.0
- Run `az bicep build -f somefile.bicep`

**Expected behavior**

Bicep build should work

**Environment summary**
Installed using `az upgrade` and repaired using the MSI. Both don't work.
Windows 10, x64. `az bicep build` command ran from a git bash terminal.

**Additional context**

**Comment by yonzhan** [comment]
route to service team

**Comment by fluffy-cakes** [comment]
I get the same when trying to run a deployment via `az deployment...`

```bash
➜  mgmt_group: az deployment tenant create -l uksouth -f main.bicep
type object 'datetime.datetime' has no attribute 'fromisoformat'

➜  mgmt_group: python3

Python 3.8.5 (default, Jan 27 2021, 15:41:15)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> from datetime import datetime
>>> datetime.fromisoformat('2021-04-13')
datetime.datetime(2021, 4, 13, 0, 0)

>>> datetime.
datetime.astimezone(        datetime.isoweekday(        datetime.timestamp(
datetime.combine(           datetime.max                datetime.timetuple(
datetime.ctime(             datetime.microsecond        datetime.timetz(
datetime.date(              datetime.min                datetime.today(
datetime.day                datetime.minute             datetime.toordinal(
datetime.dst(               datetime.month              datetime.tzinfo
datetime.fold               datetime.mro(               datetime.tzname(
datetime.fromisocalendar(   datetime.now(               datetime.utcfromtimestamp(
datetime.fromisoformat(     datetime.replace(           datetime.utcnow(
datetime.fromordinal(       datetime.resolution         datetime.utcoffset(
datetime.fromtimestamp(     datetime.second             datetime.utctimetuple(
datetime.hour               datetime.strftime(          datetime.weekday(
datetime.isocalendar(       datetime.strptime(          datetime.year
datetime.isoformat(         datetime.time(

>>> datetime._
```

I am running the deployment from a fresh build of my Docker container built this morning: https://github.com/fluffy-cakes/devops_tools/

**Comment by zhoxing-ms** [comment]
@shenglol Could you please help to have a look at this issue~

**Comment by shenglol** [comment]
@zhoxing-ms already submitted a PR: https://github.com/Azure/azure-cli/pull/17675. Can you take a look?

**Comment by shenglol** [comment]
For those who get this error, please use the temporary workaround mentioned in https://github.com/Azure/bicep/issues/2243#issuecomment-818914668 to the configure Azure CLI to use Python 3.8.

**Comment by IJsbrandVV** [comment]
Just to be clear. While a PR has been closed, no new version of the Azure CLI has been released.

Can we just use 2.22.0 now or should we wait for a patch release 2.22.1?

**Comment by fluffy-cakes** [comment]
@IJsbrandVV , i noted that too, however the PR was pushed into the Dev branch, which i assume means will be in their backlog for the next release?

`zhoxing-ms merged 1 commit into Azure:dev from shenglol:bicep-python3.6-fix 16 minutes ago`

**Comment by zhoxing-ms** [comment]
@IJsbrandVV @fluffy-cakes Yes, closing PR usually only means that the code in PR is merged into the dev branch, but the related code needs the next sprint to be released.
Maybe you can refer to the solution mentioned by shenglol as a [workaround ](https://github.com/Azure/bicep/issues/2243#issuecomment-818914668)

**Comment by zhoxing-ms** [comment]
@shenglol May I ask if this issue has a wide impact on users? If only a small number of users are affected, can they use the private package of edge build at first, and we will release it in the next sprint (05-04)?
If hotfix is necessary, what is the expected release time?

**Comment by zhoxing-ms** [comment]
If you need to launch hotfix, please use the package of edge build to do test to ensure that the quality meets the expectation. And then we'll help you release the hotfix

**Comment by IJsbrandVV** [comment]
@zhoxing-ms For now I have disabled automatic updates of the Azure CLI, uninstalled it and reinstalled 2.21.0  
  
In my personal opinion an issue is not closed unless an actual fix has been released but w/e. I've warned our team not to upgrade and wait for the next release.

**Comment by IJsbrandVV** [comment]
@zhoxing-ms as for people affected I don't have an answer. I'm just a windows dev that doesn't have a local python installation (Our codebase is C# / typescript so why would I have python installed?) and use bicep. Everyone in that profile would have this issue and I've seen more and more issues pop up regarding this, i.e. #17700 #17694 and the bicep issue [#2243](https://github.com/Azure/bicep/issues/2243) you linked already.

**Comment by zhoxing-ms** [comment]
@IJsbrandVV Hi, we released a new version (`2.22.1`) of hotfix to address this issue today. Please upgrade CLI to the latest version (`2.22.1`) and try again.

**Comment by IJsbrandVV** [comment]
That's great @zhoxing-ms 

I just updated to the latest version and everything works again. Thanks for the quick response 👍
